### PR TITLE
[12.0][ADD] pricelist_by_category_qty

### PR DESCRIPTION
**Pull request by mayjo79**
This module will add a button inside the sale order and will compute the discount not by line but by quantities of the same product category, in the same sale order.
This module was originally developed in v7.0 by the company Julius : https://github.com/julius-network-solutions/julius-openobject-addons/tree/master/pricelist_by_category_qty

### CODE REVIEW

**Comment by bealdav** [comment]
Thanks @mayjo79 for your contribution.
Misc improvements required here
https://travis-ci.org/OCA/sale-workflow/jobs/657217253#L319
and here
https://travis-ci.org/OCA/sale-workflow/jobs/657217253#L370

**Comment by bealdav** [comment]
@mayjo79 don't hesitate to have a look into https://github.com/OCA/maintainer-tools/tree/master/template/module
specially for readme folder and file header, thanks

**Comment by mayjo79** [comment]
@mathieuvatel could you approve this PR ?
Thanks

**Review by mathieuvatel — APPROVED** [review]
Looks fine to me,
You can change the addon logo (icon.png) if you want, no problem.

**Comment by mayjo79** [comment]
Thanks @mathieuvatel but I prefer to keep your logo

**Comment by mathieuvatel** [comment]
Alright then. No problem.

**Comment by bealdav** [comment]
@mayjo79 PSC of this repo'll ask you to merge your fix commits with migration one.

Also you may change commit author with real author @mathieuvatel

**Comment by mayjo79** [comment]
I don't understand where the problem is. The original module in v7 was done by @mathieuvatel and the migration in v12 by myself.

File: pricelist_by_category_qty/__init__.py
Comment (rousseldenis) [inline pricelist_by_category_qty/__init__.py]: Please use limited license format for all your files like : https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L1

File: pricelist_by_category_qty/__manifest__.py
Comment (rousseldenis) [inline pricelist_by_category_qty/__manifest__.py]: Please license either AGPL or LGPL like : https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L14

File: pricelist_by_category_qty/models/sale.py
Comment (rousseldenis) [inline pricelist_by_category_qty/models/sale.py:68]: In _get_quantity_to_compute, prefer a real multi implementation as you will do as many search as lines numbers.

So, you could do it before the loop here under and build a dict of quantities per line.

File: pricelist_by_category_qty/README.rst
Comment (rousseldenis) [inline pricelist_by_category_qty/README.rst]: You can use readme fragments to easily build OCA compliant README, see: https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Change at least README and licensing formats.

File: pricelist_by_category_qty/models/sale.py
Comment (mayjo79) [reply pricelist_by_category_qty/models/sale.py:68]: I understand your request for improvement but is it really necessary? I took over an existing module and I do not develop often enough to support this refactoring.

**Comment by bealdav** [comment]
Please @mayjo79 could you execute these cmd

`git reset --soft HEAD~9`
`git add .`
`git commit -m "MIGR v12 pricelist_by_category_qty"`
`git push -f`

Thanks

File: pricelist_by_category_qty/models/sale.py
Comment (bealdav) [reply pricelist_by_category_qty/models/sale.py:68]: I share your opinion @rousseldenis but for a first contribution this module could be merged as it with an 'Alpha' value in `development_status` key in \_\_manifest\_\_.py with a ROADMAP.rst in readme directory to indicate that module should be benefit of refactoring to work with good performance.
On my side, if I decide to use it, I'll convince my customer to pay me to optimize it.
@mayjo79 you also may add a CONTRIBUTORS.rst file (in readme) to mention you as contributors.
Thanks a lot

Comment (rousseldenis) [reply pricelist_by_category_qty/models/sale.py:68]: Ok, so include that in ROADMAP

**Review by bealdav — APPROVED** [review]

File: pricelist_by_category_qty/models/sale.py
Comment (bealdav) [reply pricelist_by_category_qty/models/sale.py:68]: All is done as expected

**Comment by bealdav** [comment]
Bonjour @rousseldenis il ne me manque plus que ta revue ici. Merci pour un dernier coup d'oeil

**Review by tonygalmiche — APPROVED** [review]
I tested locally on my computer
It's ok for me, it works

**Comment by bealdav** [comment]
Hi @rousseldenis could you finalize your review and merge this PR or close it if quality is not sufficient for you ? For an alfa, I think it's OK. While new version comes in next weeks we have to decide because not so good to have pending PR on older versions. 

Thanks to consider it.

File: pricelist_by_category_qty/README.rst
Comment (HaraldPanten) [inline pricelist_by_category_qty/README.rst:35]: I would improve the README. let's say... Do the user need to activate pricelists and/or pricelist by formula, for example?

**Review by HaraldPanten — CHANGES_REQUESTED** [review]
Just some README comments

File: pricelist_by_category_qty/README.rst
Comment (bealdav) [reply pricelist_by_category_qty/README.rst:35]: Yes it's not perfect but as a first contribution, I think we should encourage new people releasing as it. It can be improved for future versions in many corners

Comment (HaraldPanten) [reply pricelist_by_category_qty/README.rst:35]: Sure it will be, no problem.

I'm not talking about an improvement but the minimal configuration steps of the README file. Do you know what I mean?

**Comment by bealdav** [comment]
@HaraldPanten you're right. Maybe you can use github suggestion to add it, then PSC can merge with it ? Thanks

**Review by HaraldPanten — COMMENTED** [review]
@bealdav 

I do not agree but I'll not block it; as you preffer. 

IMO, the minimum configuration instructions are a must. Lots of PRs are being blocked everyday due to much less important details... 🤷‍♂️

**Review by HaraldPanten — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]
Code review. Thanks.

@HaraldPanten Don't be too rigorous. We can improve this in a furher version.

**Comment by HaraldPanten** [comment]
> Code review. Thanks.
> 
> @HaraldPanten Don't be too rigorous. We can improve this in a furher version.

It's OK. Just suggested changes 👍

**Pull request merged**
